### ISSUE TITLE
Relationship without 'object' field in notifications

### ISSUE THREAD

**Issue by ctc-nakatsuka** [Fiware/context.Orion-LD#857]
When an entity sent in a notification contains a relationship attribute has an array, the `object` field of the relationship becomes the `value` field, as shown below:

```
{
  "id": "urn:ngsi-ld:Notification:60ad9c2adb256bf6b027c3e5",
  "type": "Notification",
  "subscriptionId": "urn:ngsi-ld:Subscription:60acb735db256bf6b027c3df",
  "notifiedAt": "2021-05-26T00:54:02.447Z",
  "data": [
    {
      "id": "urn:ngsi-ld:Building:store001",
      "type": "Building",
      "foo": {
        "type": "Relationship",
        "value": [                             <--- here
          "urn:ngsi-ld:Building:store002"
        ]
      }
    }
  ]
}
```

It occurs in the following version:

```
{
  "orionld version": "0.7.0",
  "orion version":   "1.15.0-next",
  "uptime":          "0 d, 0 h, 25 m, 13 s",
  "git_hash":        "nogitversion",
  "compile_time":    "Fri Mar 19 12:26:12 UTC 2021",
  "compiled_by":     "root",
  "compiled_in":     "buildkitsandbox",
  "release_date":    "Fri Mar 19 12:26:12 UTC 2021",
  "doc":             "https://fiware-orion.readthedocs.org/en/master/"
}
```

**Comment by kzangeli** [comment]
Found the error and I'm fixing it.

Just wanted to send you a warning about the usage of Arrays for Relationships.

This is not part of the NGSI-LD standard.
The NGSI-LD standard says that a Relationship shall have an "object" that is of the type String.
- Array is not allowed.

I implemented the acceptance of Array of Strings (_made a lot of sense to me ..._) by mistake and as I now have users depending on it, I have decided to keep it until I'm able to develop a replacement.

The replacement is `datasetId`, which will be more difficult to use, but will give you the same capabilities.
I'm also working on trying to change the NGSI-LD API spec (in ETSI ISG CIM which is defining the API), as to allow for relationships to have arrays, but doesn't look as if I'm going to be successful there ...

It's gonna take some time for Orion-LD to fully support `datasetId' so, the relationship arrays will be supported for quite some time.
Just wanted to warn you about this ...
